metricbeat/module/system/service/data.go

```diff
@@ -0,0 +1,221 @@
+		childData = true
+		msData["exec_code"] = translateChild(props.ExecMainCode)
+		childProc["exit_code"] = props.ExecMainStatus
+	}
+
+	//only send timestamp if it's valid
+	if timeSince != 0 {
+		msData["state_since"] = time.Unix(0, timeSince)
+	}
+
+	//only prints PID data if we have a PID
+	if props.ExecMainPID > 0 {
+		childData = true
+		childProc["pid"] = props.ExecMainPID
+	}
+
+	if childData {
+		event.RootFields["process"] = childProc
+	}
+	event.RootFields["systemd"] = common.MapStr{
+		"unit":          unit.Name,
+		"fragment_path": props.FragmentPath,
+	}
+
+	return event, nil
+}
+
+// getMetricsFromServivce checks what accounting we have enabled and uses that to determine what metrics we can send back to the user
+func getMetricsFromServivce(props Properties) common.MapStr {
+	metrics := common.MapStr{}
+	//This error checking is because we don't _quite_ trust the maps we get back from the API.
+	if props.CPUAccounting {
+		metrics["cpu"] = common.MapStr{
+			"usage": common.MapStr{
+				"nsec": props.CPUUsageNSec,
+			},
+		}
+	}
+
+	if props.MemoryAccounting {
+		metrics["memory"] = common.MapStr{
+			"usage": common.MapStr{
+				"bytes": props.MemoryCurrent,
+			},
+		}
+	}
+
+	if props.TasksAccounting {
+		metrics["tasks"] = common.MapStr{
+			"count": props.TasksCurrent,
+		}
+	}
+
+	if props.IPAccounting {
+		metrics["network"] = common.MapStr{
+			"in": common.MapStr{
+				"packets": props.IPIngressPackets,
+				"bytes":   props.IPIngressBytes,
+			},
+			"out": common.MapStr{
+				"packets": props.IPEgressPackets,
+				"bytes":   props.IPEgressBytes,
+			},
+		}
+	}
+
+	return metrics
+}
+
+// translateChild translates the SIGCHILD code that systemd gets from the MainPID under its control into a string value.
+// Normally this shows up in systemctl status like this:  Main PID: 5305 (code=exited, status=0/SUCCESS)
+// This mapping of SIGCHILD int codes comes from the kernel. systemd does something similar to turn them into pretty strings
+func translateChild(code int32) string {
+	/*
+	   CLD_EXITED: 1
+	   CLD_KILLED: 2
+	   CLD_DUMPED: 3
+	   CLD_TRAPPED: 4
+	   CLD_STOPPED: 5
+	   CLD_CONTINUED: 6
+	*/
+	switch code {
+	case 1:
+		return "exited"
+	case 2:
+		return "killed"
+	case 3:
+		return "dumped"
+	case 4:
+		return "trapped"
+	case 5:
+		return "stopped"
+	case 6:
+		return "continued"
+	default:
+		return "unknown"
+	}
+
+}
+
+// timeSince emulates the behavior of `systemctl status` with regards to reporting time:
+// "Active: inactive (dead) since Sat 2019-10-19 16:31:25 PDT;""
+// The dbus properties data contains a number of different timestamps, which one systemd reports depends on the unit state
+func timeSince(props Properties, state string) (int64, error) {
+
+	var ts uint64
+	//normally we would want to check these values before we cast,
+	//but checking the state before we access `props` insures the values will be there.
+	switch state {
+	case "active":
+		fallthrough
+	case "reloading":
```

Comment:
Nit. Instead of using `fallthrough` here, add multiple cases to the same `case`, I find it easier to read.
```suggestion
	case "active", "reloading":
```